js/modules/k6/grpc/client.go

```diff
@@ -461,7 +461,17 @@ func (c *Client) parseInvokeParams(paramsVal goja.Value) (*invokeParams, error)
 				// TODO(rogchap): Should we manage a string slice?
 				strval, ok := kv.(string)
 				if !ok {
-					return result, fmt.Errorf("metadata %q value must be a string", hk)
+					// The spec defines that Binary-valued keys end in -bin
+					//  https://grpc.io/docs/what-is-grpc/core-concepts/#metadata
+					if strings.HasSuffix(hk, "-bin") {
```

Comment:
@sapphire-janrain, thanks for the change. Generally looks good :+1: 

However, I think we might have an issue with the logical levels. We have the binary-dependent logic, under the `"string"` case. 

How about moving the change outer of the current if/else block, which belongs to the non-binary keys? I believe it also should address the TODO `// TODO(rogchap): Should we manage a string slice?`

In a pseudo-code, I meant modifying the body of the for `hk, kv := range rawHeaders`'s loop like:
```
for key, value {
   if key is binary {
      // try to cast value to a binary and store strval
      // if can't cast binary return an error
   } else {
      // try to cast to a string and store it
      // if can't cast to a string, return an error
   }

   // put the string representation into map
}
```

In that way, it has fewer nested layers, and we could make the error message even more precise since we know that the specific key is either binary or string.

How does that all sound to you?

Reply:
This is part `1` - yes

but `2` is more about the fact we still actually make the binary metadata into a string one - which likely will lose "preceision".

Reply:
:thinking:  right, `2` sounds like a valid concern

Reply:
I don't think we should *require* `-bin` keys to be binary as that's ultimately a breaking change. People likely have existing work-arounds in their JS which work for their use cases, and this is supposed to be a simple change.

Reply:
Seems to me like `-bin` means that it wil be binary. So us threating it as anything different is not great.

>  ultimately a breaking change

Okay, but it is a breaking change that fixes a thing we do not support to something that we do support in a good way. 

Which ultimately makes it a bugfix.

> People likely have existing work-arounds in their JS which work for their use cases

People *can't* have workarounds for all cases - so there will be users who will need this to be binary and they will nto have a way to fix this. 

Also ... I guess this never worked until this point, so it won't even be a thing users could've actually did.

Reply:
It currently works for any data in the low end, so if someone's existing binary data only used < 0x80 it could have passed a string none the wiser. I'm not against the change exactly it's just something to consider. But yeah there certainly aren't any workarounds for *all* cases, just some.